agent-inject/agent/annotations.go

```diff
@@ -35,6 +35,10 @@ const (
 	// If not provided, a default generic template is used.
 	AnnotationAgentInjectTemplate = "vault.hashicorp.com/agent-inject-template"
 
+	// AnnotationAgentInjectToken is the annotation key for injecting the token
+	// from auth/token/lookup-self
+	AnnotationAgentInjectToken = "vault.hashicorp.com/agent-inject-token-only"
```

Comment:
I wonder if `vault.hashicorp.com/agent-inject-token-only` is confusing: it suggests that we're _only_ going to render a token.  Maybe this should be renamed to `agent-inject-share-token`?

Reply:
How about just `agent-inject-token`?

Reply:
Works for me.

Reply:
Added in d9dcbcddb2d915921112e0e98258fec3052bce40